front/lib/resources/group_resource.ts

```diff
@@ -23,8 +26,27 @@ import { GroupVaultModel } from "@app/lib/resources/storage/models/group_vaults"
 import { GroupModel } from "@app/lib/resources/storage/models/groups";
 import type { ReadonlyAttributesType } from "@app/lib/resources/storage/types";
 import { getResourceIdFromSId, makeSId } from "@app/lib/resources/string_ids";
+import type { ResourceFindOptions } from "@app/lib/resources/types";
+import { ResourceError } from "@app/lib/resources/types";
 import { UserResource } from "@app/lib/resources/user_resource";
 
+export type GroupErrorType =
+  | "user_not_found"
+  | "user_not_member"
+  | "user_already_member"
+  | "system_or_global_group";
+
+export class GroupError extends Error {
```

Comment:
I'm not sure we want to create one error type per resource. I'm worried that this scales pretty badly. We could probably create one base error class that accepts a code and a message, it could be enough, right?